### PR TITLE
Add tableau server and visualization configuration to project settings

### PR DESCRIPTION
**Pull request by stephherbers**
## Summary
Ticket [here](https://dimagi-dev.atlassian.net/browse/USH-1025)
Note this is currently branching off this pr: https://github.com/dimagi/commcare-hq/pull/29910

## Feature Flag

This is under the EMBEDDED_TABLEAU feature flag added in the linked PR above

## Product Description

For each domain, we are assuming one tableau server but more than one visualizations from that server.

Tableau Server Config:
Screen Shot 2021-06-21 at 12 02 59 PM

Tableau Visualization Config with a list view of all the visualizations where you can delete them and then the screen to add a new one:
Screen Shot 2021-06-21 at 12 10 59 PM
Screen Shot 2021-06-21 at 12 11 12 PM
Screen Shot 2021-06-21 at 12 11 28 PM
Screen Shot 2021-06-21 at 12 11 41 PM

## Safety Assurance

- [X] Risk label is set correctly
- [X] All migrations are backwards compatible and won't block deploy
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [X] If QA is part of the safety story, the "Awaiting QA" label is used
- [X] I have confidence that this PR will not introduce a regression for the reasons below

### QA Plan

I will not be requesting QA

### Safety story

Able to test adding, removing, and modifying of the configurations locally and soon on staging.

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/apps/reports/templates/reports/tableau_visualization_detail.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{% extends "hqwebapp/base_section.html" %}
```
Comment (orangejenny) [inline corehq/apps/reports/templates/reports/tableau_visualization_detail.html:1]: Seems like this and `tableau_server.html` could be the same template. Could you extract a template called something like `crispy_form.html` or `single_crispy_form.html` and stick it in the `hqwebapp` django app?

There are about 20 other files that could probably take advantage of such a template (pasting here for future reference, not asking you to update them):
corehq/apps/accounting/templates/accounting/accounts_base.html
corehq/apps/accounting/templates/accounting/plans_base.html
corehq/apps/accounting/templates/accounting/subscriptions_base.html
corehq/apps/accounting/templates/accounting/test_reminder_emails.html
corehq/apps/accounting/templates/accounting/trigger_accounting_tests.html
corehq/apps/accounting/templates/accounting/trigger_bookkeeper.html
corehq/apps/accounting/templates/accounting/trigger_customer_invoice.html
corehq/apps/accounting/templates/accounting/trigger_invoice.html
corehq/apps/case_migrations/templates/case_migrations/migration.html
corehq/apps/domain/templates/domain/admin/calendar_fixture.html
corehq/apps/domain/templates/domain/admin/commtrack_settings.html
corehq/apps/domain/templates/domain/admin/location_fixture.html
corehq/apps/domain/templates/domain/admin/openclinica_settings.html
corehq/apps/domain/templates/domain/admin/project_privacy.html
corehq/apps/domain/templates/domain/admin/transfer_domain.html
corehq/apps/integration/templates/integration/dialer_settings.html
corehq/apps/integration/templates/integration/gaen_otp_server_settings.html
corehq/apps/integration/templates/integration/hmac_callout_settings.html
corehq/apps/locations/templates/locations/filter_and_download.html
corehq/apps/ota/templates/ota/prime_restore_cache.html
corehq/apps/sms/templates/sms/subscribe_sms.html
corehq/apps/sso/templates/sso/accounting_admin/new_identity_provider.html
corehq/motech/dhis2/templates/dhis2/dataset_map_create.html

File: corehq/apps/reports/views.py [hunk 2]
```diff
@@ -2120,3 +2123,131 @@ def project_health_user_details(request, domain, user_id):
         'groups': ', '.join(g.name for g in Group.by_user_id(user_id)),
         'submission_by_form_link': submission_by_form_link,
     })
+
+
+class TableauServerView(BaseProjectReportSectionView):
```
Comment (orangejenny) [inline corehq/apps/reports/views.py:2128]: As long as the feature flag exists, these views should check it. They should also check user permissions. I think these views should check `edit_motech`, to be similar to the `BiometricIntegrationView`, another integrations-type view, but product should confirm that. It looks like the current integration views are a bit inconsistent. It's possible we should have stricter permissions for the USH integrations like `GaenOtpServerSettingsView`, which are currently open to all users.

The final flag & permissions logic should also be applied in `tabclasses.py`.

**Review by orangejenny — COMMENTED** [review]
It's great how quickly you pulled this together. My feedback is minor, though the question of checking the flag & verifying what permissions to use is blocking.

File: corehq/apps/reports/views.py (hunk 2 above)
Comment (stephherbers) [reply corehq/apps/reports/views.py:2128]: Thank you for flagging this! Confirmed that the feature flag **_is_** needed for now, but still awaiting the user permissions answer.

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/reports/views.py (hunk 2 above)
Comment (stephherbers) [reply corehq/apps/reports/views.py:2128]: _Update_: having these setting under a feature flag with no other user permissions checks is sufficient for the foreseeable future

File: corehq/apps/reports/templates/reports/tableau_visualization_detail.html (hunk 1 above)
Comment (stephherbers) [reply corehq/apps/reports/templates/reports/tableau_visualization_detail.html]: Created template in this commit: https://github.com/dimagi/commcare-hq/pull/29946/commits/4cde39f69f84ffe870464060e05a4e23ebe3260f

**Comment by orangejenny** [comment]
🥳

**Pull request merged**